test/lib/jdk/test/lib/hotspot/ir_framework/ArgumentValue.java

```diff
@@ -0,0 +1,310 @@
+/*
+ * Copyright (c) 2021, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package jdk.test.lib.hotspot.ir_framework;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Method;
+import java.lang.reflect.Parameter;
+import java.util.Random;
+
+/**
+ * This class represents an argument value specified by {@link Argument} in {@link Arguments}.
+ */
+class ArgumentValue {
+    private static final Random random = new Random();
+
+    private final Object argumentValue;
+    private final boolean isToggleBoolean;
+    private final boolean isRandomEach;
+    private final boolean isFixedRandom;
+    private final Class<?> randomClass;
+    private boolean previousBoolean;
+
+    private ArgumentValue(Object argumentValue, Boolean booleanToggle, boolean isFixedRandom) {
```

Comment:
Why is `booleanToggle` `Boolean` and `isFixedRandom` `boolean`?

Reply:
A hack to have a third state (`true`, `false`, and `null`) without introducing a separate class for it.